### PR TITLE
Use @param and @return in all interface fns

### PR DESCRIPTION
**Pull request by ajsantander**

### CODE REVIEW

File: protocol/synthetix/contracts/interfaces/IMarketManagerModule.sol [hunk 1]
```diff
@@ -5,82 +5,121 @@ pragma solidity ^0.8.0;
...
+    function getMarketTotalDebt(uint128 marketId) external view returns (int256);
 
     /**
      * @notice Returns the total collateral for the specified market.
+     * @param marketId The id of the market whose collateral is being queried.
+     * @return The market's total collateral, denominated with 18 decimals of precision.
      */
-    /// @notice gets the snxUSD value of the collateral backing this market.
-    function getMarketCollateral(uint128 marketId) external view returns (uint);
+    function getMarketCollateral(uint128 marketId) external view returns (uint256);
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IMarketManagerModule.sol:109]: This is a confusing name. It's more like `getMarketCredit` or `getMarketCreditCapacity`, but it doesn't include market-provided collateral. This should probably be clarified in the comments, but I think we should rename as well. We also have `getMarketCollateralAmount` in the Market Collateral Module, which adds to the confusion.

I don't have a great suggestion off the top of my head. Let's discuss further?

File: protocol/synthetix/contracts/interfaces/IPoolConfigurationModule.sol [hunk 2]
```diff
@@ -5,34 +5,53 @@ pragma solidity ^0.8.0;
...
 
     /**
      * @notice Un-marks a pool as preferred by the system owner.
+     * @param poolId The id of the pool that is to be no longer approved.
      */
     function removeApprovedPool(uint128 poolId) external;
 
     /**
      * @notice Retrieves the unique system preferred pool.
+     * @return The if of the pool that is currently set as preferred in the system.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IPoolConfigurationModule.sol:48]:
```suggestion
     * @return The id of the pool that is currently set as preferred in the system.
```

File: protocol/synthetix/contracts/interfaces/IPoolConfigurationModule.sol [hunk 3]
```diff
@@ -5,34 +5,53 @@ pragma solidity ^0.8.0;
...
      */
-    function getPreferredPool() external view returns (uint);
+    function getPreferredPool() external view returns (uint256);
 
     /**
      * @notice Retrieves the pool that are approved by the system owner.
+     * @return An Array with all the pool ids that are approved in the system.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IPoolConfigurationModule.sol:54]:
```suggestion
     * @return An array with all of the pool ids that are approved in the system.
```

File: protocol/synthetix/contracts/interfaces/IPoolModule.sol [hunk 4]
```diff
@@ -24,26 +30,38 @@ interface IPoolModule {
...
     /**
      * @notice Gets fired when pool nominee accepts nomination.
+     * @param poolId The id of the pool for which the owner nomination was accepted.
+     * @param owner The address of the new owner of the pool, which accepted the nomination.
      */
     event PoolOwnershipAccepted(uint128 indexed poolId, address indexed owner);
 
     /**
      * @notice Gets fired when pool owner revokes nomination.
+     * @param poolId The if of the pool in which the nomination was revoked.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IPoolModule.sol:40]:
```suggestion
     * @param poolId The id of the pool in which the nomination was revoked.
```

File: protocol/synthetix/contracts/interfaces/IPoolModule.sol [hunk 5]
```diff
@@ -24,26 +30,38 @@ interface IPoolModule {
...
+     * @param owner The current owner of the pool.
      */
     event PoolNominationRevoked(uint128 indexed poolId, address indexed owner);
 
     /**
      * @notice Gets fired when pool nominee renounces nomination.
+     * @param poolId The if of the pool for which the owner nomination was renounced.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IPoolModule.sol:47]:
```suggestion
     * @param poolId The id of the pool for which the owner nomination was renounced.
```

File: protocol/synthetix/contracts/interfaces/IRewardsManagerModule.sol [hunk 6]
```diff
@@ -5,23 +5,46 @@ pragma solidity ^0.8.0;
  * @title Module for connecting rewards distributors to vaults.
  */
 interface IRewardsManagerModule {
+    /**
+     * @notice Emitted when the pool owner or an existing reward distributor sets up rewards for vault participants.
+     * @param poolId The id of the pool on which rewards were distributed.
+     * @param collateralType The collateral type of the pool on which rewards were distributed.
+     * @param distributor The reward distributor associated to the rewards that were distributed.
+     * @param amount The amount of rewards that were distributed, denominated with 18 decimals of precision.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IRewardsManagerModule.sol:13]:
```suggestion
     * @param amount The amount of rewards that were distributed.
```

I think the amount is agnostic to decimal precision. It just gets passed through to the payout function.

File: protocol/synthetix/contracts/interfaces/IRewardsManagerModule.sol [hunk 7]
```diff
@@ -5,23 +5,46 @@ pragma solidity ^0.8.0;
...
+        uint256 duration
     );
 
+    /**
+     * @notice Emitted when a vault participant claims rewards.
+     * @param accountId The id of the account that claimed the rewards.
+     * @param poolId The id of the pool where the rewards were claimed.
+     * @param collateralType The address of the collateral used in the pool's rewards.
+     * @param distributor The address of the rewards distributor associated with these rewards.
+     * @param amount The amount of rewards that were claimed, denominated with 18 decimals of precision.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IRewardsManagerModule.sol:32]:
```suggestion
     * @param amount The amount of rewards that were claimed.
```

File: protocol/synthetix/contracts/interfaces/IRewardsManagerModule.sol [hunk 8]
```diff
@@ -39,40 +65,59 @@ interface IRewardsManagerModule {
 
     /**
      * @notice Called by pool owner or an existing distributor to set up rewards for vault participants.
+     * @param poolId The id of the pool to distribute rewards to.
+     * @param collateralType The address of the collateral used in the pool's rewards.
+     * @param amount The amount of rewards to be distributed, denominated with 18 decimals of precision.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IRewardsManagerModule.sol:70]:
```suggestion
     * @param amount The amount of rewards to be distributed.
```

File: protocol/synthetix/contracts/interfaces/IRewardsManagerModule.sol [hunk 9]
```diff
@@ -39,40 +65,59 @@ interface IRewardsManagerModule {
...
         uint32 duration
     ) external;
 
     /**
      * @notice Allows a user with appropriate permissions to claim rewards associated with a position.
+     * @param poolId The id of the pool to claim rewards on.
+     * @param collateralType The address of the collateral used in the pool's rewards.
+     * @param accountId The id of the account that is to claim the rewards.
+     * @param distributor The address of the rewards distributor associated with the rewards being claimed.
+     * @return The amount of rewards that were available for the account and thus claimed, denominated with 18 decimals of precision.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IRewardsManagerModule.sol:88]:
```suggestion
     * @return The amount of rewards that were available for the account and thus claimed.
```

File: protocol/synthetix/contracts/interfaces/IUSDTokenModule.sol [hunk 10]
```diff
@@ -8,16 +8,23 @@ import "@synthetixio/core-modules/contracts/interfaces/ITokenModule.sol";
  */
 interface IUSDTokenModule is ITokenModule {
     /**
-     * @notice Allows the core system to burn stablecoins with transfer allowance.
+     * @notice Allows the core system to burn stablecoins held by the `from` address, provided that it has given allowance to `spender`.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IUSDTokenModule.sol:11]:
```suggestion
     * @notice Allows the core system to burn snxUSD held by the `from` address, provided that it has given allowance to `spender`.
```

File: protocol/synthetix/contracts/interfaces/IUSDTokenModule.sol [hunk 11]
```diff
@@ -8,16 +8,23 @@ import "@synthetixio/core-modules/contracts/interfaces/ITokenModule.sol";
...
+     * @param from The address that holds the tokens to be burnt.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IUSDTokenModule.sol:12]:
```suggestion
     * @param from The address that holds the snxUSD to be burned.
```

File: protocol/synthetix/contracts/interfaces/IUSDTokenModule.sol [hunk 12]
```diff
@@ -8,16 +8,23 @@ import "@synthetixio/core-modules/contracts/interfaces/ITokenModule.sol";
...
+     * @param spender The address to which the holder has given allowance to.
+     * @param amount The amount of tokens to be burnt, denominated with 18 decimals of precision.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IUSDTokenModule.sol:14]:
```suggestion
     * @param amount The amount of snxUSD to be burned, denominated with 18 decimals of precision.
```

File: protocol/synthetix/contracts/interfaces/IVaultModule.sol [hunk 13]
```diff
@@ -34,46 +45,66 @@ interface IVaultModule {
...
+    ) external returns (uint256);
 
     /**
      * @notice Returns the debt of the specified staking position. Credit is expressed as negative debt.
      * @dev This is not a view function, and actually updates the entire debt distribution chain.
      * @dev To call this externally as a view function, use `staticall`.
+     * @param accountId The id of the account being queried.
+     * @param poolId The id of the pool in which the account's position is held.
+     * @param collateralType The address of the collateral used in the queried position.
+     * @return The amount of USD debt held by the position, denominated with 18 decimals of precision.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IVaultModule.sol:74]:
```suggestion
     * @return The amount of debt held by the position, denominated with 18 decimals of precision.
```

File: protocol/synthetix/contracts/interfaces/IVaultModule.sol [hunk 14]
```diff
@@ -34,46 +45,66 @@ interface IVaultModule {
...
         uint128 accountId,
         uint128 poolId,
         address collateralType
-    ) external returns (uint);
+    ) external returns (uint256);
 
     /**
      * @notice Returns the debt of the specified staking position. Credit is expressed as negative debt.
      * @dev This is not a view function, and actually updates the entire debt distribution chain.
      * @dev To call this externally as a view function, use `staticall`.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IVaultModule.sol:70]:
```suggestion
     * @dev Call this function using `callStatic` to treat it as a view function.
```

File: protocol/synthetix/contracts/interfaces/IVaultModule.sol [hunk 15]
```diff
@@ -34,46 +45,66 @@ interface IVaultModule {
...
+    ) external view returns (uint256 collateralAmount, uint256 collateralValue);
 
     /**
      * @notice Returns all information pertaining to a specified staking position in the vault module.
+     * @param accountId The id of the account being queried.
+     * @param poolId The id of the pool in which the account's position is held.
+     * @param collateralType The address of the collateral used in the queried position.
+     * @return collateralAmount The amount of collateral used in the position, denominated with 18 decimals of precision.
+     * @return collateralValue The value of the collateral used in the position, denominated with 18 decimals of precision.
+     * @return debt The amount of USD debt held in the position, denominated with 18 decimals of precision.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IVaultModule.sol:106]:
```suggestion
     * @return debt The amount of debt held in the position, denominated with 18 decimals of precision.
```

File: protocol/synthetix/contracts/interfaces/IVaultModule.sol [hunk 16]
```diff
@@ -82,37 +113,47 @@ interface IVaultModule {
...
+            uint256 collateralAmount,
+            uint256 collateralValue,
+            int256 debt,
+            uint256 collateralizationRatio
         );
 
     /**
      * @notice Returns the total debt (or credit) that the vault is responsible for. Credit is expressed as negative debt.
      * @dev This is not a view function, and actually updates the entire debt distribution chain.
      * @dev To call this externally as a view function, use `staticall`.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/interfaces/IVaultModule.sol:125]:
```suggestion
     * @dev Call this function using `callStatic` to treat it as a view function.
```

**Review by noahlitvin — CHANGES_REQUESTED** [review]

File: protocol/synthetix/contracts/interfaces/IMarketManagerModule.sol (hunk 1 above)
Comment (ajsantander) [reply protocol/synthetix/contracts/interfaces/IMarketManagerModule.sol:109]: Will track in a separate PR

**Review by noahlitvin — APPROVED** [review]

**Pull request merged**